src/engines/html5/html5.js

```diff
@@ -246,6 +246,76 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     }
   }
 
+  /**
+   * this function runs on the players text tracks and checks if there is already a text track with the same language
+   * as the new one.
+   * @param {PKTextTrack} textTrack - the text track to check
+   * @returns {number} - the index of the text track with the same language (if there is any). and -1 if there isn't
+   * @public
+   */
+  getLanguageTrackIndex(textTrack: PKTextTrack): number {
```

Comment:
We dont need this API. You can follow this logic in the external captions handler by calling player.getVideoElement().textTracks